Sources/SpeziLLMOpenAI/FunctionCalling/LLMFunctionParameterArray.swift

```diff
@@ -25,10 +26,16 @@ import Foundation
 ///     /// Manual conformance to `LLMFunctionParameterArrayElement` of a custom array item type.
 ///     struct CustomArrayItemType: LLMFunctionParameterArrayElement {
 ///         static let itemSchema: LLMFunctionParameterItemSchema = .init(
-///             type: .object,
-///             properties: [
-///                 "firstName": .init(type: .string, description: "The first name of the person"),
-///                 "lastName": .init(type: .string, description: "The last name of the person")
+///             "type": "object",
+///             "properties": [
+///                 "firstName": [
+///                     "type": "string",
+///                     "description": "The first name of the person")
+///                 ],
```

Comment:
I am wondering if we can add a nicely typed type for this instead of a dictionary; it can always map to a dictionary under the hood. Would be cool to avoid loosing that type-safe element?

Reply:
Previously, SpeziLLMOpenAI wrapped around the Swift types provided by the `OpenAI` package, which would then eventually be passed to the API. 
With the OpenAI OpenAPI spec, such types aren't generated, but the JSON schemas are instead validated for correctness as they're being encoded in the [OpenAPIObjectContainer](https://swiftpackageindex.com/apple/swift-openapi-runtime/1.5.0/documentation/openapiruntime/openapiobjectcontainer) type.

Introducing such wrapper types again would require precise alignment with the OpenAI, which would make it, I could imagine, harder to maintain over time.
I could imagine that’s one reason why the [official OpenAI Python package](https://github.com/openai/openai-python), which is also generated from the OpenAI OpenAPI specification, does not offer wrapper types either, AFAICT.

What do you think?

Reply:
I think adding an extension initializer/function that takes the well-typed arguments of one wants to use them would be beneficial and would avoid issues with string keys that are not correct or malformatted. Still allowing to pass in a dictionary might be an escape hatch that we can still provide. The OpenAPI surface is quite stable and if we use e.g. an enum for the type of the parameter can also have an `other` case with an associated string value.

Reply:
I most definitely agree with @PSchmiedmayer here, I would follow the typical Swift paradigm and provide as much type safety as possible. 
As mentioned by @PSchmiedmayer, I would implement well-typed inits / functions etc. that then map to the underlying `String` dictionary. And yes, an escape hatch that passes the raw dictionary might be beneficial!

Reply:
Just to be clear and because I'm a little clueless about how to implement this easily: the only way I'm seeing here is to re-implement these two types from the MacPaw OpenAI package that SpeziLLM was using previously, and then reverting the changes around the initialisers accordingly.

https://github.com/StanfordSpezi/SpeziLLM/blob/26b1e07756556b1b329f3a9a3267a5f99c0c2e78/Sources/SpeziLLMOpenAI/FunctionCalling/LLMFunctionParameterWrapper.swift#L12-L15

I was not able to find a definitive documentation for the fileds that the OpenAI API accepts here, including the ones that are currently support by SpeziLLM, e.g., `minItems`, `maxItems`, `uniqueItems`.

The function calling documentation mentions none of them: https://platform.openai.com/docs/guides/function-calling

What do you think?